### DISCUSSION TITLE
Build and Run with MPI

### DISCUSSION THREAD

**Discussion by jphaupt** [fortran-lang/fpm#492] [General]
I came across a [discussion on discourse](https://fortran-lang.discourse.group/t/why-should-i-use-cmake/953/26) where other users were trying to build an MPI program with fpm. However, the solutions there do not work for me. Namely, having 
```
[build]
external-modules = "mpi"  # (if using)
```
in my `fpm.toml` file results in the error `Key external-modules is not allowed in [build]`. I guess maybe the issue is that this update hasn't been pushed to release? My conda installation is up to date. 

The other solution (hacky but would be nice to have these sorts of workarounds) was 
```
fpm build --compiler 'gfortran' --flag '$(mpif90 --showme:link)'
```
but this simply gives the error `Unable to find source for module dependency: "mpi" used by "app/piMC_mpi.F90"`, even though I can build with mpifort or indeed with `gfortran piMC_mpi.F90 $(mpif90 --showme:link)`.

**Comment by awvwgk** [comment]
The *external-modules* key has been added with https://github.com/fortran-lang/fpm/pull/438, but this is not yet part of a released version, unfortunately.

**Reply by jphaupt** [reply]
That's what I figured, thanks. This is a useful thread for me to read as well. Is the hacky workaround also only compatible with this change? I thought this would be a solution in the mean time, but it looks like not.

**Reply by jphaupt** [reply]
The build mentioned in [this thread](https://github.com/fortran-lang/fpm/discussions/493) solved this problem as well (or at least gave me a workaround).
